synapse/handlers/auth.py

```diff
@@ -1851,15 +1860,24 @@ def verify_short_term_login_token(self, token: str) -> LoginTokenAttributes:
         user_id = get_value_from_macaroon(macaroon, "user_id")
         auth_provider_id = get_value_from_macaroon(macaroon, "auth_provider_id")
 
+        oidc_sid: Optional[str] = None
+        try:
+            oidc_sid = get_value_from_macaroon(macaroon, "oidc_sid")
+        except MacaroonVerificationFailedException:
+            pass
+
         v = pymacaroons.Verifier()
         v.satisfy_exact("gen = 1")
         v.satisfy_exact("type = login")
         v.satisfy_general(lambda c: c.startswith("user_id = "))
         v.satisfy_general(lambda c: c.startswith("auth_provider_id = "))
+        v.satisfy_general(lambda c: c.startswith("oidc_sid = "))
```

Comment:
If I'm remembering how macaroons work -- this will be fine if there's no `oidc_sid` also?